### PR TITLE
Add new `Style/IndetHeredoc` cop

### PR DESCRIPTION
**Pull request by pocke**
This cop checks the following code.

```ruby
# bad
s = <<-END
This is a text.
This is a text.
END

# good
s = <<-END.strip_indent
  This is a text.
  This is a text.
END

# It is not bad, because the indentation may be intended.
s = <<-END
      This is a text.
      This is a text.
END

# It is bad, because if `<<~` is used, the indentation is not included to a string.
s = <<~END
      This is a text.
      This is a text.
END
```

-----------------

Before submitting the PR make sure the following are checked:

* [x] Wrote [good commit messages][1].
* [ ] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Used the same coding conventions as the rest of the project.
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Added an entry to the [Changelog](../blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](../blob/master/CONTRIBUTING.md#changelog-entry-format).
* [x] All tests are passing.
* [x] The new code doesn't generate RuboCop offenses.
* [x] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences.
* [x] Updated cop documentation with `rake generate_cops_documentation` (required only when you've added a new cop or changed the configuration/documentation of an existing cop).

[1]: http://chris.beams.io/posts/git-commit/

### CODE REVIEW

**Comment by bbatsov** [comment]
Great addition! 👍 

One of those days I should continue with all my ideas for powerpack as well. :D

File: lib/rubocop/cop/style/indent_heredoc.rb [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+      #   something
+      #   END
+      #
+      #   # good
+      #   <<~END
+      #     something
+      #   END
+      #
+      #   # good
+      #   <<~END.strip_heredoc
```
Comment (deivid-rodriguez) [inline lib/rubocop/cop/style/indent_heredoc.rb]: Would it be `<<-` here?

File: lib/rubocop/cop/style/indent_heredoc.rb [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+          powerpack: 'strip_indent'
+        }.freeze
+
+        def on_str(node)
+          return unless heredoc?(node)
+
+          body_indent_level = body_indent_level(node)
+
+          if heredoc_indent_type(node) == '~'
+            expected_indent_level = base_indnet_level(node) + indentation_width
```
Comment (deivid-rodriguez) [inline lib/rubocop/cop/style/indent_heredoc.rb]: Typo in method call and definition `base_indnet_level` -> `base_indent_level`

File: lib/rubocop/cop/style/indent_heredoc.rb [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+      #   # good
+      #   <<~END.strip_heredoc
+      #     something
+      #   END
+      class IndentHeredoc < Cop
+        include ConfigurableEnforcedStyle
+
+        MSG = 'Use %d spaces for indentation in a heredoc, ' \
+              'relative to the start of the line where the ' \
+              'heredoc delimiter is.'.freeze
```
Comment (deivid-rodriguez) [inline lib/rubocop/cop/style/indent_heredoc.rb]: Should the message be improved with the specific actions the user need to take? I'm thinking of the ruby 2.2 case, where auto-correct with the default configuration won't work.

**Comment by deivid-rodriguez** [comment]
Great addition. I left some (late) comments in the PR.

**Comment by pocke** [comment]
@deivid-rodriguez Thanks for your comment! I'll create a new PR 👍

**Pull request merged**